Code/Framework/AzToolsFramework/AzToolsFramework/ComponentModes/BoxComponentMode.h

```diff
@@ -21,6 +21,9 @@ namespace AzToolsFramework
     {
     public:
         AZ_CLASS_ALLOCATOR_DECL
+        AZ_RTTI(BoxComponentMode, "{8E09B2C1-ED99-4945-A0B1-C4AFE6FE2FA9}", EditorBaseComponentMode)
```

Comment:
We're reflecting the classes so that we can use rtti to retrieve all descendant classes of `EditorBaseComponentMode`. I checked `AZ_TYPE_INFO` and it doesn't have the same signature as `AZ_RTTI` that would let us specify a base class; I also looked into `AZ_TYPE_INFO_SPECIALIZE` but that also seems to cover a different case (specializing a base type).

Unless I'm missing something, `AZ_TYPE_INFO` doesn't seem to cover this case.

Reply:
Correct, `AZ_TYPE_INFO` is only for associating a Uuid with the class.
`AZ_RTTI` is needed to establish a class hierarchy between the Uuids.

Reply:
Ah okay I misunderstood the use case, `AZ_RTTI` is fine here, it's just quite often it's used when it's not needed